test/puppetlabs/trapperkeeper/services/metrics/metrics_service_test.clj

```diff
@@ -58,7 +69,24 @@
          (is (= 200 (:status resp)))
          (doseq [[metric path] body
                  :let [resp (http-client/get (str "http://localhost:8180/metrics/v1" path))]]
-           (is (= 200 (:status resp))))))
+           (is (= 200 (:status resp)))))
+
+       (let [resp (http-client/get "http://localhost:8180/metrics/v2/list")
+             body (parse-response resp)]
+         (is (= 200 (:status  resp)))
+         (doseq [[namesp mbeans] (get body "value") mbean (keys mbeans)
+                 :let [url (str "http://localhost:8180/metrics/v2/read/"
+                                (jolokia-encode (str namesp ":" mbean))
+                                ;; NOTE: Some memory pools intentionally don't
+                                ;; implement MBean attributes. This results
+                                ;; in an error being thrown when those
+                                ;; attributes are read and is expected.
+                                "?ignoreErrors=true")
+                       resp (http-client/get url)
+                       body (parse-response resp)]]
+           ;; NOTE: Jolokia returns 200 OK for most responses. The actual
+           ;; status code is in the JSON payload that makes up the body.
+           (is (= 200 (get body "status"))))))
```

Comment:
Yeah, assuming the purpose of this is just to confirm that the metrics listed can all return "success" on an attempt to read, this makes sense.